nilearn/reporting/glm_reporter.py

```diff
@@ -763,28 +763,38 @@ def _make_stat_maps_contrast_clusters(
         components_template_text = html_template_obj.read()
     for contrast_name, stat_map_img in stat_img.items():
         component_text_ = string.Template(components_template_text)
-        thresholded_stat_map, threshold = threshold_stats_img(
+
+        # Only use threshold_stats_img to adjust the threshold
+        # that we will pass to  _clustering_params_to_dataframe
+        # and _stat_map_to_svg
+        # Necessary to avoid :
+        # https://github.com/nilearn/nilearn/issues/4192
+        _, threshold = threshold_stats_img(
```

Comment:
Actually you cannot do that: you need to get the thresholdedimage because you want to take into account the cluster_threshold that is provided. In the implementation you propose, this gets forgotten.